### PR TITLE
Update Institutions model

### LINKED ISSUE #472: Merge new and old Institution classes
Once #452 is merged, we should replace the previous [`hmda.model.fi.Institution`](https://github.com/cfpb/hmda-platform/blob/master/model/src/main/scala/hmda/model/fi/Institution.scala) with the new `hmda.model.institution.Institution` version.

### PR DESCRIPTION
**Pull request by nickgrippin**
Still working on the JSON spray stuff

Closes #472

### CODE REVIEW

File: api/src/test/scala/hmda/api/model/ModelGenerators.scala [hunk 1]
```diff
@@ -70,4 +78,41 @@ trait ModelGenerators {
...
+  }
+
+  implicit def institutionTypeGen: Gen[InstitutionType] = {
+    Gen.oneOf(
+      Bank,
+      CreditUnion,
+      SavingsAndLoan,
+      NonDepositInstType,
+      NoDepositTypeInstType
+    )
```
Comment (lgdean) [inline api/src/test/scala/hmda/api/model/ModelGenerators.scala]: I prefer `Gen.oneOf(InstitutionType.values)` (assuming it works) so I won't have to remember to modify it in #438.
Same deal for `Agency`, `ExternalIdType`, and any others I missed.

File: persistence/src/main/scala/hmda/persistence/institutions/InstitutionPersistence.scala [hunk 2]
```diff
@@ -27,8 +27,13 @@ object InstitutionPersistence {
         case InstitutionCreated(i) =>
           InstitutionsState(institutions + i)
         case InstitutionModified(i) =>
-          val x = institutions.find(x => x.id == i.id).getOrElse(Institution())
-          InstitutionsState((institutions - x) + i)
+          if (institutions.exists(x => x.id == i.id)) {
+            val x = institutions.find(x => x.id == i.id).get
+            InstitutionsState((institutions - x) + i)
+          } else {
+            InstitutionsState(institutions + i)
```
Comment (lgdean) [inline persistence/src/main/scala/hmda/persistence/institutions/InstitutionPersistence.scala]: I suspect this line is untested. (Check codecov results when available.)

I'm also not too happy about iterating over the list twice and using `get`.  Are you aiming to improve readability?  If so, I suggest:
```
          val others: Set[Institution] = institutions.filterNot(_.id == i.id)
          InstitutionsState(others + i)
```
We may eventually want to use something that supports fast+easy lookup by id, too.

No strong preference about the `x => x.id` vs `_.id` here, by the way.

File: persistence/src/main/scala/hmda/persistence/institutions/InstitutionPersistence.scala [hunk 3]
```diff
@@ -73,7 +78,7 @@ class InstitutionPersistence extends PersistentActor with ActorLogging {
       }
 
     case GetInstitutionById(institutionId) =>
-      val institution = state.institutions.find(x => x.id == institutionId).getOrElse(Institution())
+      val institution = state.institutions.find(x => x.id.toString == institutionId)
```
Comment (lgdean) [inline persistence/src/main/scala/hmda/persistence/institutions/InstitutionPersistence.scala]: Note: our choice of `Int` for the ID is not set in stone (see point 3.i [here](https://github.com/cfpb/hmda-platform/pull/452#issuecomment-236234651)), so feel free to keep using `String` if it makes your changes easier, or seems better for some other reason.  I realize you'll have to change some code either way, though.

File: persistence/src/main/scala/hmda/persistence/institutions/InstitutionPersistence.scala (hunk 2 above)
Comment (nickgrippin) [inline persistence/src/main/scala/hmda/persistence/institutions/InstitutionPersistence.scala]: This is definitely more readable, thanks!

**Comment by jmarin** [comment]
Overall, LGTM
I'm surprised that `Institution` id is now an `Int`. That will not work with [`LEI`](https://www.gleif.org/en/lei-focus/about-the-lei/iso-17442-the-lei-code-structure) in the future. Since it's a topic that is not related to this PR, it shouldn't hold it back

**Pull request merged**